src/device/device_ledger.cpp

```diff
@@ -473,21 +528,26 @@ namespace hw {
         this->length_send = offset;
         this->exchange();
 
-        //clear key
-        memmove(ledger::viewkey.data,  this->buffer_recv+64, 32);
-        memmove(ledger::spendkey.data, this->buffer_recv+96, 32);
-        unlock_device();
-      }catch (...)  {
-        unlock_device();
-        throw;
-      }
-      #endif
-      return true;
+        //View key is retrievied, if allowed, to speed up blockchain parsing
+        memmove(this->viewkey.data,  this->buffer_recv+0,  32);
```

Comment:
I don't like that the argument is named `viewkey` while the `device_ledger` class also has the member data of the same name, which is confusing and can cause bugs in the future. Please consider renaming either of them.